postgis/lwgeom_btree.c

```diff
@@ -131,10 +132,64 @@ PG_FUNCTION_INFO_V1(lwgeom_hash);
 Datum lwgeom_hash(PG_FUNCTION_ARGS)
 {
 	GSERIALIZED *g1 = PG_GETARG_GSERIALIZED_P(0);
+
 	int32_t hval = gserialized_hash(g1);
 	PG_FREE_IF_COPY(g1, 0);
 	PG_RETURN_INT32(hval);
 }
 
+static int
+lwgeom_cmp_abbrev(Datum x, Datum y, SortSupport ssup)
+{
+	/* Empty is a special case */
+	if (x == 0 || y == 0 || x == y)
+		return 0;
+	else if (x > y)
+		return 1;
+	else
+		return -1;
+}
 
+static int
+lwgeom_cmp_full(Datum x, Datum y, SortSupport ssup)
+{
+	GSERIALIZED *g1 = (GSERIALIZED *)PG_DETOAST_DATUM(x);
+	GSERIALIZED *g2 = (GSERIALIZED *)PG_DETOAST_DATUM(y);
+	int ret = gserialized_cmp(g1, g2);
+	return ret;
+}
 
+static bool
+lwgeom_abbrev_abort(int memtupcount, SortSupport ssup)
+{
+	return 0;
+}
+
+static Datum
+lwgeom_abbrev_convert(Datum original, SortSupport ssup)
+{
+	GSERIALIZED *g = (GSERIALIZED *)PG_DETOAST_DATUM(original);
+	return gserialized_get_sortable_hash(g);
```

Comment:
Same as `lwgeom_cmp_full`. If the context outlives the comparison we might need to free the detoasted datum manually to avoid memory issues; but I don't know if it does :D

Reply:
fixed